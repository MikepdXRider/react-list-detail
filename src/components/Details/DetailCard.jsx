// Presentational component
// Recieves state, props and functions from container component
// Displays selected list item details.

import React from 'react'

export default function DetailCard({ name, photoUrl, hair, weapon, affiliation, profession, first }) {
    return (
        <article>
            <h3>{name}</h3>
            <img src={`${photoUrl}`} alt={`${name}`} />
            <p>Affiliation: {affiliation}</p>
            <p>Hair: {hair}</p>
            <p>Weapon: {weapon}</p>
            <p>Profession: {profession}</p>
            <p>First: {first.length ? first : 'None'}</p>
        </article>
    )
}
