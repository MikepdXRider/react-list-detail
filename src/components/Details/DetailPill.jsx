import React from 'react'
import styles from './Details.css'

export default function DetailPill({title, content}) {
    return (
        <div className={styles.pillContainer}>
            <div className={`${styles.pillText} ${styles.pillTitleArea}`}>{title.toUpperCase()}</div>
            <div className={`${styles.pillText} ${styles.pillContentArea}`}>{content && content !=='"' ? content : 'None Listed'}</div>
        </div>
    )
}
