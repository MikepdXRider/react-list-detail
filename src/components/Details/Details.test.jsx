import { render } from "@testing-library/react"
import DetailCard from "./DetailCard.jsx"
import DetailPill from "./DetailPill.jsx"

const testObj = {
    name: 'test-name',
    photoUrl: 'test-photo',
    hair: 'test-hair',
    weapon: 'test-weapon',
    affiliation: 'test-affilitation',
    profession: 'test-profession',
    first: 'test-first'
}

describe('details presentational component snapshots', () => {
    it('detail card snapshot', () => {
        const {container} = render(
            <DetailCard 
            {...testObj}   
            />
        )

        expect(container).toMatchSnapshot();
    })

    it('detail pill snapshot', () => {
        const {container} = render(
            <DetailPill title='test-title' content={testObj.hair} />
        )

        expect(container).toMatchSnapshot();
    })
})