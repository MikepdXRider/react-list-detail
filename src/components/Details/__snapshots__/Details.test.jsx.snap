// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details presentational component snapshots detail card snapshot 1`] = `
<div>
  <section
    class="detailsContainer"
  >
    <img
      alt="test-name"
      class="detailsImage"
      src="test-photo"
    />
    <div
      class="detailsContChild"
    >
      <div
        class="pillContainer"
      >
        <div
          class="pillText pillTitleArea"
        >
          NAME
        </div>
        <div
          class="pillText pillContentArea"
        >
          test-name
        </div>
      </div>
      <div
        class="pillContainer"
      >
        <div
          class="pillText pillTitleArea"
        >
          PROF
        </div>
        <div
          class="pillText pillContentArea"
        >
          test-profession
        </div>
      </div>
      <div
        class="pillContainer"
      >
        <div
          class="pillText pillTitleArea"
        >
          AFFILIATION
        </div>
        <div
          class="pillText pillContentArea"
        >
          test-affilitation
        </div>
      </div>
    </div>
    <div
      class="detailsContChild"
    >
      <div
        class="pillContainer"
      >
        <div
          class="pillText pillTitleArea"
        >
          WEAPON
        </div>
        <div
          class="pillText pillContentArea"
        >
          test-weapon
        </div>
      </div>
      <div
        class="pillContainer"
      >
        <div
          class="pillText pillTitleArea"
        >
          HAIR
        </div>
        <div
          class="pillText pillContentArea"
        >
          test-hair
        </div>
      </div>
      <div
        class="pillContainer"
      >
        <div
          class="pillText pillTitleArea"
        >
          FIRST SEEN
        </div>
        <div
          class="pillText pillContentArea"
        >
          test-first
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`details presentational component snapshots detail pill snapshot 1`] = `
<div>
  <div
    class="pillContainer"
  >
    <div
      class="pillText pillTitleArea"
    >
      TEST-TITLE
    </div>
    <div
      class="pillText pillContentArea"
    >
      test-hair
    </div>
  </div>
</div>
`;
